modules/core/shared/src/main/scala/lucuma/core/model/ElevationRange.scala

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model
+
+import cats._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Interval
+import lucuma.core.optics.SplitEpi
+import monocle.Focus
+import monocle.Prism
+import monocle.macros.GenPrism
+import eu.timepit.refined.api.RefinedTypeOps
+
+sealed trait ElevationRange extends Product with Serializable
+
+object ElevationRange {
+
+  final case class AirMass private (
+    min: AirMass.DecimalValue,
+    max: AirMass.DecimalValue
+  ) extends ElevationRange
+
+  final case class HourAngle private (
+    minHours: HourAngle.DecimalHour,
+    maxHours: HourAngle.DecimalHour
+  ) extends ElevationRange
```

Comment:
We don't need the `sealed abstract case class` trick to prevent using the `copy` constructor anymore correct?

Reply:
Correct. And we can still use the `copy` method within the companion object, eg from lenses.